pygmt/clib/session.py

```diff
@@ -717,7 +717,9 @@ def _check_dtype_and_dim(self, array, ndim):
         """
         # check the array has the given dimension
         if array.ndim != ndim:
-            raise GMTInvalidInput(f"Expected a numpy 1d array, got {array.ndim}d.")
+            raise GMTInvalidInput(
+                f"Expected a numpy {ndim}-D array, got {array.ndim}-D."
+            )
 
         # check the array has a valid/known data type
```

Comment:
Upper-case beginning for consistency, together with the suggestion by @michaelgrund.
```suggestion
        # Check that the array has a valid/known data type
```